core/spec/models/spree/promotion/rules/user_role_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'spec_helper'
+
+describe Spree::Promotion::Rules::UserRole, type: :model do
+  let(:rule) { Spree::Promotion::Rules::UserRole.new }
+  let(:special_role) { create(:role, name: :special) }
+  let(:roles) { create_list(:role, 3) }
+  let(:user) { create(:user, spree_roles: [special_role]) }
+
+  context '#eligible?(order)' do
+    let(:order) { Spree::Order.new }
+
+    it 'should not be eligible if order don\'t have a user' do
+      allow(rule).to receive_messages(roles: roles.append(special_role))
```

Comment:
These tests are nice and feature-complete, but we strive to avoid having stubs in our specs, as it can make them brittle.

It should be possible to write this without stubbing https://gist.github.com/jhawthorn/74b9a554a307675d69e624db5cef33f3